Framework/Algorithms/src/SolidAngle.cpp

```diff
@@ -89,37 +180,54 @@ void SolidAngle::exec() {
 
   const auto &spectrumInfo = inputWS->spectrumInfo();
   const auto &detectorInfo = inputWS->detectorInfo();
-  const Kernel::V3D samplePos = spectrumInfo.samplePosition();
-  g_log.debug() << "Sample position is " << samplePos << '\n';
 
-  const int loopIterations = m_MaxSpec - m_MinSpec;
+  // this is the solid angle of the pixel at 2theta=0
+  // this is used only if Method != GenericShape
+  double pixelAreaZero = 0.;
```

Comment:
If taking l2 from the pixel, I think this name does not make sense anymore, should rather be just `pixelArea`.